src/components/views/context_menus/ThreadListContextMenu.tsx

```diff
@@ -42,11 +50,27 @@ const contextMenuBelow = (elementRect: DOMRect) => {
     return { left, top, chevronFace };
 };
 
-const ThreadListContextMenu: React.FC<IProps> = ({ mxEvent, permalinkCreator, onMenuToggle }) => {
-    const [optionsPosition, setOptionsPosition] = useState(null);
-    const closeThreadOptions = useCallback(() => {
-        setOptionsPosition(null);
-    }, []);
+export const RovingThreadListContextMenu: React.FC<IProps> = (props) => {
+    const [onFocus, isActive, ref] = useRovingTabIndex();
+
+    return <ThreadListContextMenu
+        {...props}
+        onFocus={onFocus}
+        tabIndex={isActive ? 0 : -1}
+        inputRef={ref}
+    />;
+};
+
+const ThreadListContextMenu: React.FC<IExtendedProps> = ({
+    mxEvent,
+    permalinkCreator,
+    onMenuToggle,
+    onFocus,
+    inputRef,
+    ...props
+}) => {
+    const [menuDisplayed, _ref, openMenu, closeThreadOptions] = useContextMenu();
```

Comment:
What does `ref` do in [`useContextMenu`](https://github.com/matrix-org/matrix-react-sdk/blob/0bb7f3d3ee3d8dd8095c2ae5cc75753f2da82353/src/components/structures/ContextMenu.tsx#L520)?

For [`useRovingTabIndex`](https://github.com/matrix-org/matrix-react-sdk/blob/0bb7f3d3ee3d8dd8095c2ae5cc75753f2da82353/src/accessibility/RovingTabIndex.tsx#L257), I can see that it has utility to reference the element to register and set focus.

Reply:
It's just a utility, 90% of context menus need the ref for relative positioning. This just accommodates the general case. I might move the ref to the last in the tuple so its easier to not use.

Reply:
👍 It's currently confusing that with `useContextMenu`, it's optional to assign to the element because conversely with `useRovingTabIndex`, it seems strictly necessary to reference and work correctly.

And since `useContextMenu` and `useRovingTabIndex` both use very similar patterns, it's confusing to make it optional in one case but not the other.

Reply:
Maybe it would be better to drop `ref` from `useContextMenu` entirely. It's totally separate from the usage with `contextMenuBelow`

Reply:
Happy if that makes it clearer for everyone else, putting it last would make it effectively optional but happy to defer to the most common opinion

Reply:
In any case would be better to do as a refactor outside of this PR I hope